api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/PdqPatientFetcher.java

```diff
@@ -0,0 +1,69 @@
+package org.openmrs.module.registrationcore.api.mpi.pixpdq;
+
+import ca.uhn.hl7v2.model.Message;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiException;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiPatient;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiPatientFetcher;
+import org.openmrs.module.registrationcore.api.mpi.openempi.PatientIdentifierMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.Map;
+import org.springframework.util.CollectionUtils;
+
+// TODO add javadocs
+public class PdqPatientFetcher implements MpiPatientFetcher {
+
+    @Autowired
+    @Qualifier("registrationcore.mpiPixPdqMessageUtil")
+    private PixPdqMessageUtil pixPdqMessageUtil;
+
+    @Autowired
+    @Qualifier("registrationcore.Hl7SenderHolder")
+    private Hl7SenderHolder hl7SenderHolder;
+
+    @Autowired
+    private PatientIdentifierMapper identifierMapper;
+
+    private final Log log = LogFactory.getLog(this.getClass());
+
+    @Override
+    public MpiPatient fetchMpiPatient(String patientIdentifier, String identifierTypeUuid) {
+        String mappedMpiIdentifierTypeUuid = identifierMapper.getMappedMpiIdentifierTypeId(identifierTypeUuid);
+
+        List<Map.Entry<String, String>> queryParams = new ArrayList<Map.Entry<String, String>>();
+        queryParams.add(new AbstractMap.SimpleEntry("@PID.3.1", patientIdentifier));
+        queryParams.add(new AbstractMap.SimpleEntry("@PID.3.4", mappedMpiIdentifierTypeUuid));
+
+        try {
+            Message pdqRequest = pixPdqMessageUtil.createPdqMessage(queryParams);
```

Comment:
Shouldn't this be a pix query? You are given an identifier, I'd expect looking up a patient by identifier which should return 0 or 1 results.

Reply:
What do you think of this?

Reply:
Referring to section 3.9 PIX Query, http://www.ihe.net/Technical_Framework/upload/IHE_ITI_TF_Rev8-0_Vol2a_FT_2011-08-19.pdf

the response will be a "list of corresponding patient identifiers, if any" but what we want for this is the patient and all the patient's demographic data